easybuild/tools/containers.py

```diff
@@ -0,0 +1,305 @@
+
+    # if there is osdependencies in easyconfig then add them to Singularity recipe
+    install_os_deps = ''
+    for ec in easyconfigs:
+        for osdep in ec['ec']['osdependencies']:
+            if isinstance(osdep, basestring):
+                install_os_deps += "yum install -y %s\n" % osdep
+            # tuple entry indicates multiple options
+            elif isinstance(osdep, tuple):
+                install_os_deps += "yum --skip-broken install %s\n" % ' '.join(osdep)
+            else:
+                raise EasyBuildError("Unknown format of OS dependency specification encountered: %s", osdep)
+
+    # module names to load in container environment
+    mod_names = [e['ec'].name + '/' + det_full_ec_version(e['ec']) for e in easyconfigs]
+
+    # name of Singularity definition file
+    img_name = build_option('container_image_name')
+    if img_name:
+        def_file_label = os.path.splitext(img_name)[0]
+    else:
+        def_file_label = mod_names[0].replace('/', '-')
+
+    def_file = 'Singularity.%s' % def_file_label
+
+    # adding all the regions for writing the  Singularity definition file
+    content = SINGULARITY_TEMPLATE % {
+        'bootstrap': bootstrap_agent,
+        'from': bootstrap_from,
+        'install_os_deps': install_os_deps,
+        'easyconfigs': ' '.join(os.path.basename(e['spec']) for e in easyconfigs),
+        'mod_names': ' '.join(mod_names),
+    }
+    def_path = os.path.join(cont_path, def_file)
+
+    if os.path.exists(def_path):
+        if build_option('force'):
+            print_msg("WARNING: overwriting existing container recipe at %s due to --force" % def_path)
+        else:
+            raise EasyBuildError("Container recipe at %s already exists, not overwriting it without --force", def_path)
+
+    write_file(def_path, content)
+    print_msg("Singularity definition file created at %s" % def_path, log=_log)
+
+    return def_path
+
+
+def build_singularity_image(def_path):
+    """Build Singularity container image by calling out to 'singularity' (requires admin privileges!)."""
+
+    cont_path = container_path()
+    def_file = os.path.basename(def_path)
+
+    # use --imagename if specified, otherwise derive based on filename of recipe
+    img_name = build_option('container_image_name')
+    if img_name is None:
+        # definition file Singularity.<app>-<version, container name <app>-<version>.<img|simg>
+        dot_idx = def_file.find('.')
+        img_name = def_file[dot_idx+1:]
+
+    cmd_opts = ''
+
+    image_format = build_option('container_image_format')
+
+    # squashfs image format (default for Singularity)
+    if image_format in [None, CONT_IMAGE_FORMAT_SQUASHFS]:
+        img_path = os.path.join(cont_path, img_name + '.simg')
+
+    # ext3 image format, creating as writable container
+    elif image_format == CONT_IMAGE_FORMAT_EXT3:
+        img_path = os.path.join(cont_path, img_name + '.img')
+        cmd_opts = '--writeable'
```

Comment:
This is a typo it should be ``--writable`` and not ``--writeable``

I get the following error

```
[siddis14@amrndhl1157 easybuild-framework]$ eb M4-1.4.18.eb -C --container-base shub:shahzebsiddiqui/eb-singularity:centos-7.3.1611 --container-build --container-image-format=ext3 --experimental --force
== temporary log file in case of crash /tmp/eb-l2lS1C/easybuild-rMxfEQ.log
== Singularity tool found at /usr/local/bin/singularity
== Singularity version '2.4' is 2.4 or higher ... OK
== Singularity definition file created at /home/siddis14/.local/easybuild/containers/Singularity.M4-1.4.18
== Running 'sudo singularity build --writeable /home/siddis14/.local/easybuild/containers/M4-1.4.18.img /home/siddis14/.local/easybuild/containers/Singularity.M4-1.4.18', you may need to enter your 'sudo' password...
[sudo] password for siddis14:
ERROR: cmd "sudo singularity build --writeable /home/siddis14/.local/easybuild/containers/M4-1.4.18.img /home/siddis14/.local/easybuild/containers/Singularity.M4-1.4.18" exited with exit code 1 and output:
ERROR: Unknown option: --writeable
Cleaning up...
```

See documentation 

```

[siddis14@amrndhl1157 easybuild-framework]$ singularity build --help | grep writable
        writable:   Legacy writable image format
                    it is legacy or writable format)
    -w|--writable   Build a writable image (warning: deprecated due to sparse
        $ singularity exec --writable /tmp/debian apt-get install python

```

Reply:
woops, fix pushed